src/sinks/pulsar/config.rs

```diff
@@ -0,0 +1,292 @@
+use crate::{
+    codecs::EncodingConfig,
+    config::{AcknowledgementsConfig, GenerateConfig, Input, SinkConfig, SinkContext},
+    schema,
+    sinks::{
+        pulsar::sink::{healthcheck, PulsarSink},
+        Healthcheck, VectorSink,
+    },
+};
+use codecs::{encoding::SerializerConfig, TextSerializerConfig};
+use futures_util::FutureExt;
+use pulsar::authentication::oauth2::{OAuth2Authentication, OAuth2Params};
+use pulsar::error::AuthenticationError;
+use pulsar::{
+    compression, message::proto, Authentication, Error as PulsarError, ProducerOptions, Pulsar,
+    TokioExecutor,
+};
+use snafu::ResultExt;
+use value::Kind;
+use vector_common::sensitive_string::SensitiveString;
+use vector_config::configurable_component;
+use vector_core::config::DataType;
+
+/// Configuration for the `pulsar` sink.
+#[configurable_component(sink("pulsar"))]
+#[derive(Clone, Debug)]
+pub struct PulsarSinkConfig {
+    /// The endpoint to which the Pulsar client should connect to.
+    ///
+    /// The endpoint should specify the pulsar protocol and port.
+    #[serde(alias = "address")]
+    #[configurable(metadata(docs::examples = "pulsar://127.0.0.1:6650"))]
+    pub(crate) endpoint: String,
+
+    /// The Pulsar topic name to write events to.
+    #[configurable(metadata(docs::examples = "topic-1234"))]
+    pub(crate) topic: String,
+
+    /// The name of the producer. If not specified, Pulsar will generate a unique name.
+    #[configurable(metadata(docs::examples = "producer-name"))]
+    pub(crate) producer_name: Option<String>,
+
+    /// The log field name or tags key to use for the topic key.
+    ///
+    /// If the field does not exist in the log or in tags, a blank value will be used. If unspecified, the key is not sent.
+    ///
+    /// Pulsar uses a hash of the key to choose the topic-partition or uses round-robin if the record has no key.
+    #[configurable(metadata(docs::examples = "message"))]
+    #[configurable(metadata(docs::examples = "my_field"))]
+    pub(crate) partition_key_field: Option<String>,
```

Comment:
I think this can be updated to be one of the lookpath based types?

Reply:
Hmmm looking at `make_pulsar_event()` (where this and the other one you flagged are used) , it actually reads like these should be `Template` instead.
What do you think?

Reply:
I think you might be looking at the `kafka` source not the sink?

Reply:
(which, the kafka sink has the same logic used here for these fields (aka not lookup path types).

Reply:
I was looking at the `kafka` sink - they're Strings-as-paths, as they haven't been updated to be proper lookups.

Reply:
👍 attempted to address in https://github.com/vectordotdev/vector/pull/14345/commits/52880293532907cea973f1be79fcd2340590d008